### PR TITLE
[9.0][MIG] medical patient disease

### PR DESCRIPTION
**Pull request by lasley**
This PR upgrades medical_patient_disease concept to v9 & adds tests

This is dependent on
- [x] #86 
- [x] #71

### CODE REVIEW

**Comment by woodbrettm** [comment]
Ready for review when physician and pathology are merged.

**Comment by woodbrettm** [comment]
Alright this module is ready for review!

File: medical_patient_disease/models/medical_patient.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        inverse_name='patient_id',
+        string='Diseases',
+    )
+    count_disease_ids = fields.Integer(
+        compute='compute_count_disease_ids',
+        string='Diseases',
+    )
+
+    @api.multi
+    def compute_count_disease_ids(self):
```
Comment (woodbrettm) [inline medical_patient_disease/models/medical_patient.py]: Should this be prefixed with a '_' ?

File: medical_patient_disease/views/medical_patient_disease_view.xml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                </group>
+            </search>
+        </field>
+    </record>
+
+    <record id="medical_patient_disease_view_tree" model="ir.ui.view">
+        <field name="name">medical.patient.disease.tree</field>
+        <field name="model">medical.patient.disease</field>
+        <field name="arch" type="xml">
+            <tree string="Medical Patient Disease">
```
Comment (woodbrettm) [inline medical_patient_disease/views/medical_patient_disease_view.xml:97]: Just as a note I removed a lot of the fields from the tree view here as it was really crowded in the view. These were the fields I left in, which I thought were of importance. However I can add them back either as visible or invisible if needed, or change which fields I left here.

File: medical_patient_disease/README.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
+------------
+
+* Jonathan Nemry <jonathan.nemry@acsone.eu>
+* Dave Lasley <dave@laslabs.com>
+* Brett Wood <bwood@laslabs.com>
+* Parthiv Patel <parthiv@techreceptives.com>
```
Comment (woodbrettm) [inline medical_patient_disease/README.rst]: Remove duplicate name entry

**Comment by lasley** [comment]
@bmw95 - This module supersedes `medical_disease`. Please remove the old module & squash the commit so it works right. Let me know when done.

**Review by lasley — COMMENTED** [review]
Oops actually the old module should be a part of my commit in order to work right, not your's. Will need to reset your local head, squash the remove into mine, cherry pick your commit from remote, then force push.

**Comment by woodbrettm** [comment]
@lasley I don't see a medical_disease module in the repo, but I do see medical_emr and medical_emr_data.  Do you want me to remove those?

**Comment by lasley** [comment]
Oh yeah nvm `medical_pathology` partly replaced it so we removed there.

**Review by lasley — COMMENTED** [review]
LGTM thanks @bmw95 👍 

@t3ddftw please review

File: medical_patient_disease/models/medical_patient_disease.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    _name = 'medical.patient.disease'
+    _description = 'Medical Patient Disease'
+
+    treatment_description = fields.Char()
+    short_comment = fields.Char()
+    name = fields.Char(
+        compute='_compute_name',
+        store=True,
+    )
+    expire_date = fields.Datetime(
```
Comment (tedsalmon) [inline medical_patient_disease/models/medical_patient_disease.py:18]: Wouldn't it be nicer to add `string='Expiration Date'` so it looks better on the front end? The field is visible in a view.

**Review by tedsalmon — COMMENTED** [review]
Looks good for the most part, I just have one concern marked in my review :)

File: medical_patient_disease/models/medical_patient_disease.py (hunk 4 above)
Comment (lasley) [reply medical_patient_disease/models/medical_patient_disease.py:18]: Yup, especially because I just made a note to change this to `date_expire` in v10

Comment (lasley) [reply medical_patient_disease/models/medical_patient_disease.py:18]: To all of the dates if you would, actually. They all just got grep'd to date_\* for v10 & didn't really make sense anyways

Comment (woodbrettm) [reply medical_patient_disease/models/medical_patient_disease.py:18]: @t3ddftw nice catch! Added string attribute to all the date fields.

**Pull request merged**